topi/include/topi/generic/extern.h

```diff
@@ -47,6 +48,16 @@ inline Schedule schedule_extern(const Target& target, Array<Tensor> outs) {
     out_ops.push_back(t->op);
   }
   auto s = create_schedule(out_ops);
+
+  tvm::schedule::AutoInlineInjective(s);
+  for (auto out : outs) {
+    if (out->op->derived_from<ExternOpNode>()) {
+      continue;
+    }
+    Array<Tensor> new_outs = { out };
+    tvm::GenericFunc::Get("schedule_injective")(new_outs);
```

Comment:
@masahi or @vinx13 , this call seems to work as expected. It calls the correct schedule_injective function for the current target. However, the function that it calls creates its own schedule (see cuda/injective.py). This causes failures in the unit tests. This is probably why the previous implementation used a helper function, _schedule_injective.py). Can you think of any way to fix this when calling from C++?

Reply:
What's the problem with `cuda/injective.py`? Is overriding native generic from Python side a problem in your case?

Reply:
The specific error is `Direct host side access to device memory is detected in fused_nn_conv2d_multiply_add_nn_relu_1. Did you forget to bind?`. This is being hit in `tutorials/frontend/using_external_lib.py`

Reply:
I think it has to do with the fact that `schedule_injective` creates its own schedule, but I'm not totally sure.

Thanks a lot for all of the help, btw.

Reply:
You are right, we can't create a new schedule using `schedule_injective`. So the problem is that we don't want to duplicate the helper function in C++ right?

Reply:
Yes, exactly. Maybe we should create a new generic function? Or have a generic function that is overridden with more arguments? Is that possible?

Reply:
I don't think generic func can be overridden by number of args. We can register the function with `TVM_REGISTER_GLOBAL`